flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -614,64 +623,91 @@ private void determineMainSize(@FlexDirection int flexDirection, int widthMeasur
      */
     private int expandFlexItems(FlexLine flexLine, @FlexDirection int flexDirection,
             int maxMainSize, int paddingAlongMainAxis, int startIndex) {
+        int childIndex = startIndex;
```

Comment:
To check the min constraints, I think adding the following check in the first loop in the measureHorizontal/measureVertical is more simple and readable.
(after the child.measure(childWidthMeasureSpec, childHeightMeasureSpec); in line 371 for measureHorizontal 
and line 519 for measureVertical)

In the expandFlexItems, there is no chance that the size of the flex items are reduced, so min constraints violation can't be happen in expandFlexItems.
So I think only checking max constraints violation in expandFlexItems is more simple.

```java
 if (child.getMeasuredWidth() < lp.minWidth) {
                    child.measure(MeasureSpec.makeMeasureSpec(lp.minWidth, MeasureSpec.EXACTLY),
                            MeasureSpec.makeMeasureSpec(child.getMeasuredHeight(),
                                    MeasureSpec.EXACTLY));
                }
```

or for measureVertical

```java
 if (child.getMeasuredHeight() < lp.minHeight) {
                    child.measure(MeasureSpec.makeMeasureSpec(child.getMeasuredWidth(), MeasureSpec.EXACTLY),
                            MeasureSpec.makeMeasureSpec(lp.minHeight,
                                    MeasureSpec.EXACTLY));
                }
```